### PR TITLE
Is engaged myFT user endpoint

### PR DESCRIPTION
**Pull request by phamann**
This adds to @laurieboyes work to add a custom endpoint for spoor to enable decorating of events with our business logic of whether a given user is an engaged myFT user.

My refectorings:
- Add feature usage to query
- Add time-tree lookups to query
- Add polling class to query tomorrows node id

### CODE REVIEW

**Comment by phamann** [comment]
I've now completely refactored the massive query into individual ones and sent using batch in a single HTTP query. Makes them much easier to comprehend.

I've also added the correct business logic to determine whether they are engaged. But should probablu have some tests :)

File: server/v2/controllers/next/myft-engagement.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+							.reduce((a,b) => a.concat(b))
+							.reduce((a, b) => {
+								let key = Object.keys(b)[0];
+								a[key] = b[key];
+								return a;
+							}, {});
+
+			const accessedMyftSource = data.feature_usage_events_last_30_days > 0;
+			const usedPush = data.push_notifications_shown_last_7_days > 0 && data.push_notifications_clicked_last_30_days > 0;
+			const usedEmail = data.email_open_events_last_7_days > 0 && data.email_clicked_events_last_30_days;
```
Comment (laurieboyes) [inline server/v2/controllers/next/myft-engagement.js]: Could do with a > 0 bit on data.email_clicked_events_last_30_days for consistency's sake

File: server/v2/lib/tomorrow-id.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+const CronJob = require('cron').CronJob;
+const moment = require('moment');
+const db = require('../../utils/db');
+
+const tomorrowQuery = `MATCH (year:Year {value: {year}})
+			MATCH (year)-[:CHILD]->(month:Month {value: {month}})
+			MATCH (month)-[:CHILD]->(day:Day {value: {day}})
+			RETURN id(day) as id;`;
+
+class Tomorrow {
```
Comment (laurieboyes) [inline server/v2/lib/tomorrow-id.js]: This is solid but we should totally do that thing I thought of too late where we index the days by a datestring later (I'll make a tech debt card)

**Comment by laurieboyes** [comment]
:+1:

File: server/v2/lib/tomorrow-id.js (hunk 2 above)
Comment (phamann) [inline server/v2/lib/tomorrow-id.js]: Yepp, love that idea. Next challenge is can you write the migration entirely in Cypher :stuck_out_tongue_winking_eye:

**Pull request merged**